packages/studio/src/components/AddPageButton/EntityIdField.tsx

```diff
@@ -0,0 +1,125 @@
+import Select, { MultiValue } from "react-select";
+import useStudioStore from "../../store/useStudioStore";
+import StreamScopeFieldLabel from "./StreamScopeFieldLabel";
+import { ChangeEvent, useCallback, useMemo, useState } from "react";
+import classNames from "classnames";
+import getSelectCssClasses from "../../utils/getSelectCssClasses";
+
+interface Props {
+  disabled: boolean;
+  updateSelection: (selectedIds: string[]) => void;
+  selectedIds?: string[];
+}
+
+export default function EntityIdField({
+  disabled,
+  updateSelection,
+  selectedIds,
+}: Props) {
+  const [entitiesRecord, fetchEntities] = useStudioStore((store) => [
+    store.accountContent.entitiesRecord,
+    store.accountContent.fetchEntities,
+  ]);
+  const availableEntityTypes = Object.entries(entitiesRecord)
+    .filter(([_entityType, entityData]) => {
+      return Object.values(entityData).length > 0;
+    })
+    .map((entry) => entry[0]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [entityType, setEntityType] = useState<string | undefined>(
+    availableEntityTypes[0]
+  );
+  const entityOptions = useEntityOptions(entityType);
+  const selectedOptions = useMemo(
+    () => selectedIds?.map((id) => ({ value: id, label: id })),
+    [selectedIds]
+  );
+
+  const onEntityTypeChange = useCallback(
+    (e: ChangeEvent<HTMLSelectElement>) => {
+      setEntityType(e.target.value);
+    },
+    []
+  );
+
+  const onEntityIdChange = useCallback(
+    (selectedItems: MultiValue<{ value: string }>) => {
+      updateSelection(selectedItems.map((item) => item.value));
+    },
+    [updateSelection]
+  );
+
+  const onMenuScrollToBottom = useCallback(async () => {
+    if (!entityType) {
+      return;
+    }
+    const currentNumEntities = Object.keys(entitiesRecord[entityType]).length;
+    if (currentNumEntities >= entitiesRecord[entityType].totalCount) {
+      return;
+    }
+    setIsLoading(true);
+    await fetchEntities(entityType, Math.floor(currentNumEntities / 50));
```

Comment:
why do we use `Math.floor` here? I thought we'd either not need it at all, or need `Math.ceil`?

Reply:
I didnt want to pass a decimal, though maybe that works idk

if we have 10 entities loaded we want to ask for page 0 not page 1

Reply:
I thought the check above would already handle the cases where the number isn't a multiple of 50. Because if less than 50 entities were fetched, that means we should have reached the end and fetched the total number

Reply:
It's the same thing for more entities right? If we have 50 entities loaded we'd want to ask for page 1 not page 2

Reply:
I think it's a little safer to floor it even if we have another check

Reply:
Ah I didn't realize both floor and ceil returned the same for whole numbers. It shouldn't make a difference then. Since we're appending to the entities array, either way is a little risky if we didn't have the first check. With floor, we'd risk duplicating entities if a request returned less than 50, and with ceil, we'd risk skipping entities. But again, it shouldn't matter here

Reply:
I'm not sure how we should deal with this kind of thing long term. Maybe we can change entitiesRecord to store a map of entityId to EntityData instead of EntityData[] but that's still kind of dodging issues

Reply:
Not sure I follow. When might this be an issue? Do you mean in the case where we forget the first check?